internal/output/disk/disk.go

```diff
@@ -0,0 +1,103 @@
+package disk
+
+import (
+	"github.com/sherifabdlnaby/prism/pkg/types"
+	"go.uber.org/zap"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+	"sync"
+	"time"
+)
+
+//Disk struct
+type Disk struct {
+	FileName       string
+	FilePath       string
+	Transactions   chan types.Transaction
+	stopChan       chan struct{}
+	logger         zap.Logger
+	wg             sync.WaitGroup
+	FilePermission os.FileMode
+}
+
+//TransactionChan just return the channel of the transactions
+func (d *Disk) TransactionChan() chan<- types.Transaction {
+	return d.Transactions
+}
+
+//Init func Initialize the disk output plugin
+func (d *Disk) Init(config types.Config, logger zap.Logger) error {
+	d.FileName = config["filename"].(string)
+	d.FilePath = config["filepath"].(string)
+	path.Clean(d.FilePath)
+	d.FilePermission = config["permission"].(os.FileMode)
+	d.Transactions = make(chan types.Transaction, 1)
```

Comment:
Using buffered channel is not a good idea with `Transactions`, because you may send a transaction to the channel (that will get accepted and added to buffer), then you receive a stop signal, so now you'll stop working while there is a transaction that is still in the buffer! this will be lost. and whoever sent this transaction will wait for it forever. causing a deadlock at somepoint.

dummy had them because it was copy pasted but dummy was never serious you knw :"DD